### PR TITLE
Small telemetry updates from feedback

### PR DESCRIPTION
**Pull request by andrewlock**
## Summary of changes

- Add comments to generated code to make it clearer how the metrics map in `GetMetricBuffer()`
- Rename file

## Reason for change

Addresses minor feedback points from #4251 and offline from @kevingosse 

## Implementation details

Biggest change was a tweak to the source generator

## Test coverage

Covered by existing tests

### CODE REVIEW

File: tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// <copyright file="ConfigurationTelemetryCollectorV2.cs" company="Datadog">
+// <copyright file="ConfigurationTelemetry.Collector.cs" company="Datadog">
```
Comment (pierotibou) [inline tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs:1]: I find the new name weird. First a dot... Why? But mainly, why `ConfigurationTelemetry.Collector` whereas we have `MetricsTelemetryCollector`. That doesn't feel consistent.

That said I don't mind much

**Review by pierotibou — APPROVED** [review]

**Comment by andrewlock** [comment]
## Throughput/Crank Report:zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (4258)](https://github.com/DataDog/dd-trace-dotnet/tree/0c54edd4808d9750989589c41907d5668ccef448)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/d712de2ddcffc0db0f766210e3576cf195bccd6f)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)
- [benchmarks/2.31.0](https://github.com/DataDog/dd-trace-dotnet/tree/5a0f7f5a6023b3b0761baa6b1bcd53b8f3af5655)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (4258) (11.255M)   : 0, 11255094
    master (11.135M)   : 0, 11135473
    benchmarks/2.31.0 (11.240M)   : 0, 11240035
    benchmarks/2.9.0 (11.142M)   : 0, 11141767

    section Automatic
    This PR (4258) (7.955M)   : 0, 7954796
    master (7.916M)   : 0, 7915986
    benchmarks/2.31.0 (7.933M)   : 0, 7933476
    benchmarks/2.9.0 (7.994M)   : 0, 7994302

    section Trace stats
    master (7.957M)   : 0, 7957478
    benchmarks/2.31.0 (7.916M)   : 0, 7916115

    section Manual
    This PR (4258) (10.085M)   : 0, 10084597
    master (9.942M)   : 0, 9942300
    benchmarks/2.31.0 (10.110M)   : 0, 10110237

    section Manual + Automatic
    This PR (4258) (7.568M)   : 0, 7567755
    master (7.451M)   : 0, 7450854
    benchmarks/2.31.0 (7.578M)   : 0, 7578492

    section Version Conflict
    master (6.787M)   : 0, 6786986
    benchmarks/2.31.0 (6.919M)   : 0, 6918633

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (4258) (9.619M)   : 0, 9619472
    master (9.809M)   : 0, 9809234
    benchmarks/2.31.0 (9.617M)   : 0, 9616926
    benchmarks/2.9.0 (9.771M)   : 0, 9771077

    section Automatic
    This PR (4258) (6.779M)   : 0, 6779132
    master (6.829M)   : 0, 6828769
    benchmarks/2.31.0 (6.961M)   : 0, 6961441

    section Trace stats
    master (6.857M)   : 0, 6857432
    benchmarks/2.31.0 (6.876M)   : 0, 6875883

    section Manual
    This PR (4258) (8.500M)   : 0, 8500405
    master (8.549M)   : 0, 8549153
    benchmarks/2.31.0 (8.521M)   : 0, 8520906

    section Manual + Automatic
    This PR (4258) (6.659M)   : 0, 6659456
    master (6.694M)   : 0, 6694217
    benchmarks/2.31.0 (6.585M)   : 0, 6585005

    section Version Conflict
    master (5.950M)   : 0, 5949897
    benchmarks/2.31.0 (5.955M)   : 0, 5954552

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (4258) (10.363M)   : 0, 10362930
    master (10.380M)   : 0, 10380018
    benchmarks/2.31.0 (10.225M)   : 0, 10224516
    benchmarks/2.9.0 (10.477M)   : 0, 10476904

    section Automatic
    This PR (4258) (7.450M)   : 0, 7450080
    master (7.617M)   : 0, 7617253
    benchmarks/2.31.0 (7.576M)   : 0, 7576075
    benchmarks/2.9.0 (7.417M)   : 0, 7416639

    section Trace stats
    master (7.521M)   : 0, 7521398
    benchmarks/2.31.0 (7.642M)   : 0, 7641753

    section Manual
    This PR (4258) (9.187M)   : 0, 9187243
    master (9.196M)   : 0, 9196038
    benchmarks/2.31.0 (9.471M)   : 0, 9471068

    section Manual + Automatic
    This PR (4258) (7.237M)   : 0, 7237479
    master (7.328M)   : 0, 7327643
    benchmarks/2.31.0 (7.406M)   : 0, 7405929

    section Version Conflict
    master (6.686M)   : 0, 6685549
    benchmarks/2.31.0 (6.516M)   : 0, 6516153

```
```mermaid
gantt
    title Throughput Linux x64 (ASM) (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (4258) (7.462M)   : 0, 7461770
    master (7.412M)   : 0, 7411798
    benchmarks/2.31.0 (7.465M)   : 0, 7465154
    benchmarks/2.9.0 (7.712M)   : 0, 7712305

    section No attack
    This PR (4258) (2.061M)   : 0, 2061074
    master (2.107M)   : 0, 2107245
    benchmarks/2.31.0 (2.380M)   : 0, 2379510
    benchmarks/2.9.0 (3.233M)   : 0, 3232878

    section Attack
    This PR (4258) (1.805M)   : 0, 1804930
    master (1.770M)   : 0, 1770312
    benchmarks/2.31.0 (2.026M)   : 0, 2025734
    benchmarks/2.9.0 (2.558M)   : 0, 2557594

    section Blocking
    This PR (4258) (3.520M)   : 0, 3519555
    master (3.472M)   : 0, 3471878
    benchmarks/2.31.0 (4.084M)   : 0, 4084310

```

File: tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs (hunk 1 above)
Comment (andrewlock) [reply tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs:1]: Blame @kevingosse 😉 

The reason is that this class contains the `partial` `ConfigurationTelemetry` class. It's the "collector" part of the implementation. Mostly it's an artifact of the fact I changed the whole design part way through, and didn't want to rename `ConfigurationTelemetry` and have to touch every file in the codebase 😬

Comment (pierotibou) [reply tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs:1]: Thanks for the explanation. As I said I don't mind much and I'm happy to blame @kevingosse.

**Review by kevingosse — APPROVED** [review]

**Pull request merged**